### ISSUE TITLE
Update Facility Locator mobile map design for usability testing

### ISSUE THREAD

**Issue by xiongjaneg** [department-of-veterans-affairs/va.gov-cms#16855]
## User story
As a front end engineer, I need to know the requirements for building the staging prototype for Facility Locator mobile map usability testing. 

## Resources

Review or discuss with Alexis:
- [Research Draft Plan](https://app.mural.co/t/departmentofveteransaffairs9999/m/departmentofveteransaffairs9999/1701208273582/c0f537ca54bc4004f6a33473d3302dc251a7b05e?wid=0-1702579269238&sender=u25440eff886a526ece257322)
- [Draft Conversation Guide](https://app.mural.co/t/departmentofveteransaffairs9999/m/departmentofveteransaffairs9999/1701208273582/c0f537ca54bc4004f6a33473d3302dc251a7b05e?wid=5-1702680148115&sender=u25440eff886a526ece257322)

## Design
Per comments on #14912 we need to incorporate the following into the design 
- Add the [segmented button group](https://designsystem.digital.gov/components/button-group/) from USWDS.
- Make the numbers on the map interactive.
- For the location that's selected, the information should appear underneath that map.
- add TTY numbers to the results 
   - see [#13939](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/13939)
   - Jordan has shown what this could look in the Sketch file
- ~~if possible, account for scenarios in which pins are overlaid or very close together? (most common in searches in a urban location and "Community pharmacies" as the Facility type)~~ created new ticket for this
   - Example 1: Results E/F and I/J in [Spokane search for Community pharmacies](https://www.va.gov/find-locations/?address=Spokane%2C%20Washington%2099201%2C%20United%20States&bounds%5B%5D%5B%5D=-73.98106125513083&bounds%5B%5D%5B%5D=40.72311315443778&bounds%5B%5D%5B%5D=-73.96882903225695&bounds%5B%5D%5B%5D=40.73440219329626&bounds%5B%5D%5B%5D%5B%5D=-118.173511&bounds%5B%5D%5B%5D%5B%5D=46.907193&bounds%5B%5D%5B%5D%5B%5D=-116.673511&bounds%5B%5D%5B%5D%5B%5D=48.407193&context=Spokane%2C%20Washington%2099201%2C%20United%20States&facilityType=pharmacy&latitude=47.65703970236771&longitude=-117.41835725514301&page=1&radius=1&searchString=Spokane%2C%20Washington%2099201%2C%20United%20States&serviceType&bounds%5B%5D=-117.43442294136028&bounds%5B%5D=47.64385932554504&bounds%5B%5D=-117.40229156892636&bounds%5B%5D=47.67021675289038) (on desktop)
   - Example 2: Results A/B in [Search for All in network urgent care in zip code 59860](https://www.va.gov/find-locations/?page=1&address=59860&facilityType=urgent_care&serviceType&latitude=47.69053&longitude=-114.163761&radius=33&bounds%5B%5D=-114.913761&bounds%5B%5D=46.94053&bounds%5B%5D=-113.413761&bounds%5B%5D=48.44053&context=Polson%2C%20Montana%2059860%2C%20United%20States) (on desktop)
 
[Facility locator Figma file](https://www.figma.com/file/GvNVeG9GAXZaBpny7vNxZP/Facility-Locator?type=design&t=ZuYaVvEUlJd8BFox-6)

## Acceptance criteria - S102 (3 points)
- [x] Convert Sketch components to Figma components [in Figma file](https://www.figma.com/file/GvNVeG9GAXZaBpny7vNxZP/Facility-Locator?type=design&node-id=1%3A17356&mode=design&t=ek4tj6IRvh117g8R-1)
- [x] Add accessibility and design annotations to the mobile map design
- [x] Review with accessibility
-  ~~Review with @ALogsdon3~~ moved to separate ticket #16855
- ~~Review with engineers~~

## ACs - S103 ( 3 points)
- [x] Updates are made to design based on feedback
- [x] Link to the updated design is added to #14514 for when engineers can pick that ticket up

**Comment by thejordanwood** [comment]
I converted this to [a Figma file](https://www.figma.com/file/GvNVeG9GAXZaBpny7vNxZP/Facility-Locator?type=design&node-id=0%3A4407&mode=design&t=S2iBj60berb2jHvi-1). Please look at the Mobile Map page only, other pages haven't been updated to the new Figma components. 

@laflannery @ALogsdon3  I've added annotations to the mobile map design. Can you look this over and comment any other annotations that would be good to include for the prototype? I want to make sure I didn't miss anything relevant to accessibility or research.

**Comment by aklausmeier** [comment]
Things to consider:

- This research will use a staging "prototype" which means it is possible we will get participants who show up using the desktop version whether on laptop or a tablet
- Perigean research recruitment for accessibility in H/F has shown it's very difficult to recruit mobile participants with assistive technology - this may mean we have to broaden our scope to beyond just mobile
- We can discuss this in the next UX Sync

cc @ALogsdon3 @xiongjaneg @thejordanwood @mmiddaugh

**Comment by laflannery** [comment]
@thejordanwood Added comments

**Comment by ALogsdon3** [comment]
@xiongjaneg for this AC, can we make this a ticket for me next sprint? 3 points to be on the safe side:

> - [ ] Review with @ALogsdon3

I already discussed this with @thejordanwood :)

**Comment by xiongjaneg** [comment]
Sounds good, thank you. I'll add to the draft sprint plan.

**Comment by jilladams** [comment]
Since we broke review cycles out of this ticket for at least Alexis's feedback, I updated ACs in ticket body to reflect that changes after reviews will also happen next sprint. FYI @xiongjaneg @aklausmeier

**Comment by mmiddaugh** [comment]
@thejordanwood @aklausmeier two thoughts after viewing the annotated design in UX sync just now:
- for the sake of accuracy, when the facility type is "VA health", the user will select a service type from the drop down box (rather than entering in free text) - in other words,  the example shown in the design isn't a valid combination of parameters. 
- When you add the list view to this design, I'm wondering about the expected behavior if the user toggles between the list and map views (for example, what happens if the user selects a result in the list and then switches to the map tab - should that location get focus in the map?)

**Comment by thejordanwood** [comment]
@mmiddaugh @aklausmeier 

- Thanks for calling that out. I pulled from other mockups in the file that show the plain language search we want to have in the future. I've now updated the prototype annotations mockup to show the search in its current state.
- I'm interested to see how participants use the toggle in testing. We've discussed making changes to the mobile map, but none to the list view. Currently, users aren't able to select the items in the list view at all. @aklausmeier and I can discuss this further to determine what interactions we need to include in the prototype.

**Comment by thejordanwood** [comment]
@mmiddaugh @aklausmeier @xiongjaneg 

@aklausmeier and I have discussed what interactions should be included in the prototype. The summary is: 

- We're inclined to run usability testing to see if interactions between map/list view is a real problem to solve.
- While we're waiting on testing: @xiongjaneg can you open a ticket to look into the feasibility of switching between map and list views while retaining the selected location? Specifically, I think there may be a technical limitation that could prevent us from selecting a location in the map view, then switching to that selected location in list view. List view usually sorts by distance. If a user selects a location on the map that's far away, how would that be sorted in list view?

---

For more context, I did some research to see how other sites (Google, Yelp, Airbnb, and Zillow) handle interactions between list and map views. This a quick summary of what I found:

**Map to list view:** On these sites, the map selection would be lost when switching to list view. This is where I think the technical limitation may come in. All of these sites sorted their list view by distance like we currently do on FL. 

**List to map view:** These sites didn't have a clear path for a user to navigate from list to map view. Some had a "Directions" button on the list view that would switch to a map view, then give you directions from your current location. We already have a version of this with our "Get directions on Google Maps" links.

**Comment by xiongjaneg** [comment]
@thejordanwood #17174 stubbed for your additional comments

**Comment by ALogsdon3** [comment]
@thejordanwood Thanks for this comment. I am including a question about this in the [conversation guide](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/facilities/facility-locator/research/user-research/2024-01-mobile-map/conversation-guide.md#third-task-locate-chiropractor) because @mmiddaugh identified a use case where the pins are directly on top of one another, even at full zoom. Try doing the search for Task 3 on your phone and see if you can see both results at 65 Broadway or both results are 30 Broad St.

**Comment by thejordanwood** [comment]
Thanks @ALogsdon3! I see the problem now. 

I'm not sure this is a design issue and we may need to create a ticket for an engineer to look into this. Ideally, the pins would still be separated somewhat, like this Google Maps example of Dr. Michael Kolin who works at Manhattan Sports Medicine. 

Can we ask an engineer if something like that is possible on our map @xiongjaneg? I'm assuming they'll need a ticket for that work. 

![Screenshot 2024-02-13 at 11 15 36 AM](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/104526584/cd799bce-4f0d-4924-8f5c-50a95f31657b)

**Comment by xiongjaneg** [comment]
@thejordanwood I'll create a ticket for that. I think we have a similar ticket to do some other exploration, so both could likely go around the same time. Thank you!

**Comment by thejordanwood** [comment]
Reviewed this with @aklausmeier. Since engineering tickets are open to look into pins overlapping and map/list view interactions, this design ticket can close. We'll create any other design tickets as needed once we get more information from those engineering tickets. 

cc: @xiongjaneg
